fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/QueryPlanInfo.java

```diff
@@ -73,6 +75,24 @@ public QueryPlanInfo() {
         return this;
     }
 
+    /**
+     * Return TRUE if the info set is empty.
+     * @return TRUE if the info set is empty
+     */
+    public boolean isEmpty() {
+        return info.isEmpty();
+    }
+
+    /**
+     * Return the Key Set for info map.
+     * @return the Key Sey for the info map
+     */
+    @SuppressWarnings("java:S1452")
+    @Nonnull
+    public Set<QueryPlanInfoKey<?>> keySet() {
+        return info.keySet();
```

Comment:
Hmm, in general, I don't like mutable structures if they can be avoided. Returning the keyset here per se is allows modifications to happen and is breaking encapsulation.